static/js/components/widgets/MenuField.tsx

```diff
@@ -0,0 +1,434 @@
+          idToResultPath[childItem.identifier] = [
+            ...childItemPath,
+            childItemIdx
+          ]
+        })
+        tierIds = tierIds.concat(childItems.map(item => item.identifier))
+      }
+      tierIds = tierIds.slice(1)
+    }
+  }
+  return results
+}
+
+const getItemPath = (
+  items: InternalSortableMenuItem[],
+  itemIdToFind: string
+): Array<string | number> => {
+  let itemsToSearch: InternalSortableMenuItem[][] = [[...items]]
+  const parentPaths: Array<string | number>[] = [[]]
+  while (itemsToSearch.length > 0) {
+    const parentPath = parentPaths.pop() as Array<string | number>
+    const items = itemsToSearch.pop() as InternalSortableMenuItem[]
+    for (let idx = 0; idx < items.length; idx++) {
+      const item = items[idx]
+      if (item.id === itemIdToFind) {
+        return [...parentPath, idx]
+      }
+      if (item.children && item.children.length) {
+        itemsToSearch = itemsToSearch.concat([item.children])
+        parentPaths.push([...parentPath, idx, "children"])
+      }
+    }
+  }
+  return []
+}
+
+export default function MenuField(props: MenuFieldProps): JSX.Element {
+  const { name, value, onChange, collections } = props
+
+  const [menuData, setMenuData] = useState<{
+    hugoItems: HugoItem[]
+    internalItems: InternalSortableMenuItem[]
+  }>(() => {
+    const hugoItems = value || []
+    return {
+      hugoItems:     hugoItems,
+      internalItems: hugoItemsToInternal(hugoItems)
+    }
+  })
+
+  useEffect(() => {
+    const hugoItems = value || []
+    setMenuData({
+      hugoItems:     hugoItems,
+      internalItems: hugoItemsToInternal(hugoItems)
+    })
+  }, [value])
+
+  const [panelState, setPanelState] = useState<{
+    activeItemPath: Array<string | number>
+    activeItem: InternalSortableMenuItem | null
+    isVisible: boolean
+  }>({ activeItemPath: [], activeItem: null, isVisible: false })
+  const closeContentPanel = useCallback(() => {
+    setPanelState({ activeItemPath: [], activeItem: null, isVisible: false })
+  }, [setPanelState])
+  const openNewContentPanel = useCallback(() => {
+    setPanelState({ activeItemPath: [], activeItem: null, isVisible: true })
+  }, [setPanelState])
+
+  const [
+    itemToRemove,
+    setItemToRemove
+  ] = useState<InternalSortableMenuItem | null>(null)
+  const closeRemoveDialog = useCallback(() => {
+    setItemToRemove(null)
+  }, [setItemToRemove])
+
+  const updateValues = (menuItems: InternalSortableMenuItem[]) => {
+    const hugoItems = internalItemsToHugo(menuItems)
+    setMenuData({
+      internalItems: menuItems,
+      hugoItems:     hugoItems
+    })
+    // @ts-ignore
+    onChange({ target: { value: hugoItems, name } })
+  }
+
+  const startEditMenuItem = (menuItem: InternalSortableMenuItem) => {
+    setPanelState({
+      activeItemPath: getItemPath(menuData.internalItems, menuItem.id),
+      activeItem:     menuItem,
+      isVisible:      true
+    })
+  }
+
+  const removeMenuItem = () => {
```

Comment:
It looks like this isn't covered by tests. Can we add a test for removing a menu item?